### PR TITLE
Feat: Add logging when helpers are built

### PR DESCRIPTION
**Pull request by luispadron**
### Short description 📝

Simply adds extra logging to the `tuist` commands. Previously, when building a helper module we wouldn't log anything, this adds logging information for what module is being built and how long it took.

I believe this will let users better understand what is happening and also informs users that we cache building of these helpers (since it won't log if the helper is already built).

Before:

```sh
Cloning plugin from https://github.com/tuist/ExampleTuistPlugin.git @ 6fa42ca
Generating workspace TuistPluginTest.xcworkspace
Generating project TuistPluginTest
Project generated.
Total time taken: 2.366s
```

After:

```sh
Cloning plugin from https://github.com/tuist/ExampleTuistPlugin.git @ 6fa42ca
Building LocalPlugin (0.210s)
Building ExampleTuistPlugin (0.215s)
Building ProjectDescriptionHelpers (0.233s)
Generating workspace TuistPluginTest.xcworkspace
Generating project TuistPluginTest
Project generated.
Total time taken: 1.717s
```

### How to test the changes locally 🧐

```sh
swift run tuist generate --path projects/tuist/fixtures/app_with_plugins
```

### Checklist ✅

- [X] The code architecture and patterns are consistent with the rest of the codebase.
- [X] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [X] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [X] In case the PR introduces changes that affect users, the documentation has been updated.
- [X] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistSupport/Utils/Environment.swift [hunk 1]
```diff
@@ -68,7 +68,8 @@ public class Environment: Environmenting {
     init(directory: AbsolutePath, fileHandler: FileHandling) {
         self.directory = directory
         self.fileHandler = fileHandler
-        setup()
+        #warning("This is causing a crash because `fileHandler` logs which uses `Environment.shared`")
+        // setup()
```
Comment (luispadron) [inline Sources/TuistSupport/Utils/Environment.swift:72]: This issue can be a separate PR but:

This is a chicken-egg problem, it was working previously but curious to see what we should do here, maybe revert?

Comment (luispadron) [reply Sources/TuistSupport/Utils/Environment.swift:72]: The issue:

- Using `Environment.shared` requires `Environment.init` to be called.
- `Environment.init` calles `setup` which uses `fileHandler`
- `FileHandler` logs which uses `Environment.shared` internally to determine when to color output, etc.

Hence the 💥

Comment (luispadron) [reply Sources/TuistSupport/Utils/Environment.swift:72]: Ah: https://github.com/tuist/tuist/pull/3752 (:

Comment (luispadron) [inline Sources/TuistSupport/Utils/Environment.swift:72]:
```suggestion
        setup()
```

File: Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift [hunk 2]
```diff
@@ -7,19 +7,25 @@ import TuistSupport
...
+    /// Builds all the project description helpers incluing any `projectDescriptionHelperPlugins`.
+    ///
+    /// Note: Local project description helpers can import plugin helpers.
+    /// We build the plugin modules first to allow including them in the local helper modules.
     ///
     /// - Parameters:
     ///   - path: Path to the directory that contains the manifest being loaded.
     ///   - projectDescriptionPath: Path to the project description module.
     ///   - projectDescriptionHelperPlugins: List of custom project description helper plugins to include and build.
+    @available(*, deprecated, message: "TODO: Delete me")
```
Comment (luispadron) [inline Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift:19]:
```suggestion
```

**Review by pepicrft — APPROVED** [review]

File: Sources/TuistSupport/Utils/Environment.swift (hunk 1 above)
Comment (pepicrft) [reply Sources/TuistSupport/Utils/Environment.swift]: I'm finishing that one up. I'll let you know so that you can rebase afterwards.

**Comment by pepicrft** [comment]
I love this idea. Would you mind updating the changelog @luispadron ?

File: Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift [hunk 3]
```diff
@@ -164,7 +174,11 @@ public final class ProjectDescriptionHelpersBuilder: ProjectDescriptionHelpersBu
             customProjectDescriptionHelperModules: customProjectDescriptionHelperModules
         )
 
+        let timer = clock.startTimer()
         try System.shared.runAndPrint(command, verbose: false, environment: Environment.shared.manifestLoadingVariables)
+        let duration = timer.stop()
+        let time = String(format: "%.3f", duration)
+        logger.notice("Building \(name) (\(time)s)", metadata: .success)
```
Comment (danyf90) [inline Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift:181]:
```suggestion
        logger.notice("Built \(name) in (\(time)s)", metadata: .success)
```

**Review by danyf90 — APPROVED** [review]

File: projects/tuist/fixtures/app_with_plugins/Tuist/ProjectDescriptionHelpers/Project+Templates.swift [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import ExampleTuistPlugin
+import LocalPlugin
```
Comment (luispadron) [inline projects/tuist/fixtures/app_with_plugins/Tuist/ProjectDescriptionHelpers/Project+Templates.swift:2]:
```suggestion
```
